### ISSUE TITLE
Update cloudigrade to better fit IPP-12

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#565]
In GitLab by @katherine-black on Apr 18, 2019, 11:21

## Summary

As a software engineer, I want the software that I write to be compliant with IPP-12 so that I feel good about myself and the code I write.

## Acceptance Criteria

- [ ] Verify that we implement requirements around pagination dictated by IPP-12
- [ ] Verify that we use plural versions of our api object names as dictated by IPP-12
- [ ] Verify that we use major-minor version in the API URLs, not just the major version
  - Our current `v2` is not sufficient. IPP-12 dictates it must be `v2.0`.
  - IPP-12 does not recognize patch-level versions. Does this mean we will quickly roll through `v2.1`, `v2.2`, `v2.3`, etc.?
- [ ] Verify that our DRF APIs return object IDs with the class in the ID's property name, not just `id`.
  - e.g. the account API would return `account_id` instead of `id`

## Assumptions and Questions

- https://docs.google.com/document/d/1sEgro5-lXYJGsxOdxFOcyz7EHilrnpBVkHT_m2IXogk/edit#heading=h.9fuje8qfsx1f
- Pagination reference implementation in Koku:
  - https://github.com/project-koku/koku/blob/master/koku/api/common/pagination.py
- TODO split this into multiple issues that can be resolved independently in parallel.

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Apr 29, 2019, 13:52

changed title from **Update {-pagination and URL structur-}e to better fit IPP-12** to **Update {+cloudigrad+}e to better fit IPP-12**

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Apr 30, 2019, 14:55

Closing in favor of multiple split out stories:
- https://gitlab.com/cloudigrade/cloudigrade/issues/564
- https://gitlab.com/cloudigrade/cloudigrade/issues/565
- https://gitlab.com/cloudigrade/cloudigrade/issues/566
- https://gitlab.com/cloudigrade/cloudigrade/issues/567

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Apr 30, 2019, 14:55

closed
